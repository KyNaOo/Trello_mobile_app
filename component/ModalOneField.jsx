import React, {useState} from 'react';
import {Modal, StyleSheet, Text, TextInput, TouchableOpacity, View} from "react-native";

const ModalOneField = props => {
    const [actionValue, setActionValue] = useState('')

    return (
        <View>
        <Modal
            visible={props.modalVisible}
            animationType="fade"
            transparent={true}
            onRequestClose={() => props.setModalVisible(false)}
        >
            <View style={styles.modalContainer}>
                <View style={styles.modalContent}>
                    <Text style={styles.modalTitle}>Rename the Board</Text>
                    <TextInput
                        style={styles.input}
                        placeholder="Name"
                        onChangeText={(text) => setActionValue(text)}
                    />
                    <View style={styles.buttonContainer}>
                        <TouchableOpacity
                            style={[styles.button, styles.cancelButton]}
                            onPress={() => props.setModalVisible(false)}
                        >
                            <Text style={styles.buttonText}>Cancel</Text>
                        </TouchableOpacity>
                        <TouchableOpacity
                            style={[styles.button, styles.confirmButton]}
                            onPress={() => props.action(actionValue, props.id)}
                        >
                            <Text style={styles.buttonText}>Confirm</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
        </Modal>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 17,
    },
    stickyButton: {
        position: 'absolute',
        left: 0,
        right: 0,
        bottom: 0,
        backgroundColor: '#3498db',
        justifyContent: 'center',
        alignItems: 'center',
        paddingVertical: 10,
        paddingHorizontal: 15,
        elevation: 3,
    },
    modalContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
    },
    modalContent: {
        backgroundColor: 'white',
        padding: 20,
        borderRadius: 10,
        width: '80%',
    },
    modalTitle: {
        fontSize: 18,
        fontWeight: 'bold',
        marginBottom: 10,
    },
    input: {
        height: 40,
        borderColor: 'gray',
        borderWidth: 1,
        marginBottom: 10,
        paddingHorizontal: 10,
        width: '100%',
    },
    buttonContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    button: {
        flex: 1,
        paddingVertical: 10,
        borderRadius: 5,
        marginVertical: 5,
    },
    confirmButton: {
        backgroundColor: '#42b883',
        marginLeft: 5,
    },
    cancelButton: {
        backgroundColor: '#ef5a5a',
        marginRight: 5,
    },
    buttonText: {
        color: '#fff',
        fontWeight: 'bold',
        textAlign: 'center',
    },
});

export default ModalOneField;