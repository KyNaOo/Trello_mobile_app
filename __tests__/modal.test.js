import React from 'react';
import renderer from 'react-test-renderer';
import ModalOneField from "../component/ModalOneField";
import expect from "expect";

test('renderModalCorrectly', () => {
    const btn = renderer.create(<ModalOneField text={'Test Modal'} type={'normal'} />).toJSON();
    expect(btn).toMatchSnapshot();
});
