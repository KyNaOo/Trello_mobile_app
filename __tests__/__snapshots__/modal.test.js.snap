// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderModalCorrectly 1`] = `
<View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "padding": 20,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Rename the Board
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Name"
          style={
            {
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "marginBottom": 10,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#ef5a5a",
                "borderRadius": 5,
                "flex": 1,
                "marginRight": 5,
                "marginVertical": 5,
                "opacity": 1,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Cancel
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#42b883",
                "borderRadius": 5,
                "flex": 1,
                "marginLeft": 5,
                "marginVertical": 5,
                "opacity": 1,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Confirm
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
